### PR TITLE
Alignmentcheck

### PR DESCRIPTION
**Pull request by laureatian**
Fix wrong y_cb/cr_offset when the imported surface's fourcc is VA_FOURCC_411P.
Add memory alignment check for imported surface.

### CODE REVIEW

File: src/i965_drv_video.c [hunk 1]
```diff
@@ -1378,6 +1431,10 @@ i965_suface_external_memory(VADriverContextP ctx,
         obj_surface->cb_cr_width = obj_surface->orig_width / 2;
         obj_surface->cb_cr_height = obj_surface->orig_height / 2;
         obj_surface->cb_cr_pitch = memory_attibute->pitches[1];
+        if (tiling)
```
Comment (yakuizhao) [inline src/i965_drv_video.c]: @xhaihao , the above check is added for the NV12/P010.  It is reasonable.
The code in line 1377 is to check the tiling attribute for I420/IYUV surface earlier.

**Comment by xhaihao** [comment]
case VA_FOURCC_I420:
    case VA_FOURCC_IYUV:
    case VA_FOURCC_IMC3:
    case VA_FOURCC_I010:
        ASSERT_RET(memory_attibute->num_planes == 3, VA_STATUS_ERROR_INVALID_PARAMETER);
        ASSERT_RET(memory_attibute->pitches[1] == memory_attibute->pitches[2], VA_STATUS_ERROR_INVALID_PARAMETER);

        obj_surface->subsampling = SUBSAMPLE_YUV420;
        obj_surface->y_cb_offset = obj_surface->height;
        obj_surface->y_cr_offset = memory_attibute->offsets[2] / obj_surface->width;
        obj_surface->cb_cr_width = obj_surface->orig_width / 2;
        obj_surface->cb_cr_height = obj_surface->orig_height / 2;
        obj_surface->cb_cr_pitch = memory_attibute->pitches[1];
        if (tiling)
            ASSERT_RET(IS_ALIGNED(obj_surface->cb_cr_pitch,128),VA_STATUS_ERROR_INVALID_PARAMETER);
        else
ASSERT_RET(IS_ALIGNED(obj_surface->cb_cr_pitch,i965->codec_info->min_linear_wpitch),VA_STATUS_ERROR_INVALID_PARAMETER);

**Comment by xhaihao** [comment]
The code is added for I420/IYUV/IMC3/I010

File: src/i965_drv_video.c [hunk 2]
```diff
@@ -1378,6 +1431,10 @@ i965_suface_external_memory(VADriverContextP ctx,
...
+            ASSERT_RET(IS_ALIGNED(obj_surface->cb_cr_pitch,128),VA_STATUS_ERROR_INVALID_PARAMETER);
+        else
+            ASSERT_RET(IS_ALIGNED(obj_surface->cb_cr_pitch,i965->codec_info->min_linear_wpitch),VA_STATUS_ERROR_INVALID_PARAMETER);
```
Comment (yakuizhao) [inline src/i965_drv_video.c]: @xhaihao , the mentioned check is also used the IMC3 fourcc which supports the tiling format.
Even when the I420 fourcc is used,  it is harmless.  It has no side effect.

**Pull request merged**